### PR TITLE
Adds the spawn controller

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

SSspawning

## Why It's Good For The Game

Automated AI and mob spawning

## Changelog
:cl:
refactor: Adds the managed spawner controller
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/spawning.dm [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+/// Holder datum for various data relating to individual spawners
+/datum/spawnerdata
+	var/fire_increment = 0
+	var/required_increment
+	var/spawntypes
+	var/spawnamount
+	var/max_allowed_mobs
+	var/list/spawnedmobs = list()
+	var/datum/callback/post_spawn_cb
```
Comment (Hughgent) [inline code/controllers/subsystem/spawning.dm:103]: autodoc new vars

**Review by Hughgent — CHANGES_REQUESTED** [review]

File: code/controllers/subsystem/spawning.dm (hunk 1 above)
Comment (TiviPlus) [reply code/controllers/subsystem/spawning.dm:103]: Holders dont need autodoc

Comment (Hughgent) [reply code/controllers/subsystem/spawning.dm:103]: They do

Comment (TiviPlus) [reply code/controllers/subsystem/spawning.dm:103]: No they don't

Comment (TiviPlus) [reply code/controllers/subsystem/spawning.dm]: Done

File: code/modules/ai/spawners/xeno.dm [hunk 2]
```diff
@@ -0,0 +1,108 @@
+/obj/effect/ai_node/spawner/xeno
+	name = "Xeno AI spawner"
+
+//////Generic xeno spawners
+////////Tier ones
+
+/obj/effect/ai_node/spawner/xeno/runnergang	// RUNNNER GANG RUNNER GANG
```
Comment (SplinterGP) [inline code/modules/ai/spawners/xeno.dm:7]: that's a rouny

File: code/modules/ai/spawners/xeno.dm [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	spawndelay = 10 SECONDS
+	maxamount = 10
+
+/obj/effect/ai_node/spawner/xeno/tiertwos
+	spawntypes = list(/mob/living/carbon/xenomorph/hunter/ai, /mob/living/carbon/xenomorph/warrior/ai, /mob/living/carbon/xenomorph/spitter/ai)
+	spawnamount = 4
+	spawndelay = 10 SECONDS
+	maxamount = 10
+
+////////////////////Tier 4s
```
Comment (SplinterGP) [inline code/modules/ai/spawners/xeno.dm:69]:
```suggestion
////////////////////Tier 3s
```

**Review by SplinterGP — CHANGES_REQUESTED** [review]

**Comment by SplinterGP** [comment]
checks died with self commit, rip

**Pull request merged**